posts/onboarding.md

```diff
@@ -0,0 +1,526 @@
+        TRANSFORMS = {
+            exp.Array: lambda self, e: f"[{self.expressions(e)}]",
+        }
+
+        # Specifies how AST nodes representing data types should be converted into SQL
+        TYPE_MAPPING = {
+            exp.DataType.Type.TINYINT: "INT64",
+            exp.DataType.Type.SMALLINT: "INT64",
+            ...
+        }
+```
+
+
+Even though this example is a fairly realistic starting point, we strongly encourage you to study existing dialect implementations to understand how their various components can be modified.
+
+
+## Schema
+Previous sections described the SQLGlot components used for transpilation. This section and the next describe components needed to implement other SQLGlot functionality like column-level lineage.
+
+A Schema represents the structure of a database schema, including the tables/views it contains and their column names and data types.
+
+The file `schema.py` defines the abstract classes `Schema` and `AbstractMappingSchema`; the implementing class is `MappingSchema`.
+
+A `MappingSchema` object is defined with a nested dictionary, as in this example:
+
+```Python
+schema = MappingSchema({"t": {"x": "int", "y": "datetime"}})
+```
+
+The keys of the top-level dictionary are table names (`t`), the keys of `t`’s dictionary are its column names (`x` and `y`), and the values are the column data types (`int` and `datetime`).
+
+A schema provides information required by other SQLGlot modules (most importantly, the optimizer and column level lineage).
+
+Schema information is used to enrich ASTs with actions like replacing stars (`SELECT *`) with the names of the columns being selected from the upstream tables.
+
+## Optimizer
+
+The optimizer module in SQLGlot (`optimizer.py`) is responsible for producing canonicalized and efficient SQL queries by applying a series of optimization rules.
+
+Optimizations include simplifying expressions, removing redundant operations, and rewriting queries for better performance.
+
+The optimizer operates on the abstract syntax tree (AST) returned by the parser, transforming it into a more compact form while preserving the semantics of the original query.
+
+> [!NOTE]
+> The optimizer does not currently support statistics, so it cannot employ cost-based optimizations such as join reordering
+>
+
+
+### Optimization rules
+The optimizer essentially applies [a list of rules](https://sqlglot.com/sqlglot/optimizer.html) in a sequential manner, where each rule takes in an AST and produces a canonicalized and optimized version of it.
+
+> [!WARNING]
+> The rule order is important, as some rules depend on others to work properly. We discourage manually applying individual rules, which may result in erroneous or unpredictable behavior.
+
+The first, foundational optimization task is to _canonicalize_ an AST, which is required by other optimization steps. These rules implement canonicalization:
```

Comment:
Hmm right, the terms "canonicalize", "normalize" etc are used in various places around SQLGlot so it can definitely be confusing.. Would "standardize" be the next best verb here?